crates/matrix-sdk/tests/integration/room/beacon/mod.rs

```diff
@@ -232,26 +232,20 @@ async fn test_most_recent_event_in_stream() {
 
     let mut timeline_events = Vec::new();
 
+    let f = EventFactory::new();
     for nth in 0..25 {
-        timeline_events.push(sync_timeline_event!({
-            "content": {
-                "m.relates_to": {
-                    "event_id": "$15139375514XsgmR:localhost",
-                    "rel_type": "m.reference"
-                },
-                "org.matrix.msc3488.location": {
-                    "uri": format!("geo:{nth}.9575274619722,12.494122581370175;u={nth}")
-                },
-                "org.matrix.msc3488.ts": 1_636_829_458
-            },
-            "event_id": format!("$event_for_stream_{nth}"),
-            "origin_server_ts": 1_636_829_458,
-            "sender": "@example2:localhost",
-            "type": "org.matrix.msc3672.beacon",
-            "unsigned": {
-                "age": 598971
-            }
-        }));
+        timeline_events.push(
+            f.beacon(
+                owned_event_id!("$15139375514XsgmR:localhost"),
+                format!("geo:{nth}.9575274619722,12.494122581370175;u={nth}"),
+                Some(MilliSecondsSinceUnixEpoch(1_636_829_458u32.into())),
+            )
+            .event_id(<&EventId>::try_from(format!("$event_for_stream_{nth}").as_str()).unwrap())
```

Comment:
Putting the string or the event ID into a variable at the top might be a bit less gnarly. 

```rust
let event_id: &EventId = <&EventId>::try_from(format!(...));
```

Reply:
This is a fair suggestion, I think inlining the entire thing was a bit much, I'll split it up

Reply:
Done in fixup commit eb9ae84